### PR TITLE
nice-dcv-client: init at 2020.2.1737-1

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change

This is a client for a remote desktop protocol. I've been using this package on my NixOS machine for a few weeks, and thought it would be better to submit it here. I'm not sure that I've done the expression in the best way or with the best style, so any feedback would be appreciated.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/remote/nice-dcv-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    mv $out/usr/bin/dcvviewer $out/bin/dcvviewer
+    sed -i "s#basedir=/usr#basedir=$out/usr#" $out/bin/dcvviewer
+    ln -s $out/usr/share $out/share
+    ${glib.dev}/bin/glib-compile-schemas $out/share/glib-2.0/schemas
+    patchShebangs $out
+  '';
+
+  meta = with stdenv.lib; {
+    description = "NICE DCV is a high-performance remote display protocol that provides customers with a secure way to deliver remote desktops and application streaming from any cloud or data center to any device, over varying network conditions";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/nice-dcv-client/default.nix:64]: This is long enough for the long description. Can you add a short one?

File: pkgs/applications/networking/remote/nice-dcv-client/default.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{ stdenv, fetchurl, patchelf, bash,  glib
+  , libX11, gst_all_1, sqlite, epoxy, pango, cairo, gdk-pixbuf, e2fsprogs, libkrb5, libva, openssl
+  , pcsclite, gtk3, libselinux, libxml2
+  , python3Packages
+  , cpio
+}:
+
+stdenv.mkDerivation rec {
+  name = "nice-dcv-client-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/nice-dcv-client/default.nix:9]:
```suggestion
  pname = "nice-dcv-client";
```

File: pkgs/applications/networking/remote/nice-dcv-client/default.nix [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    ${glib.dev}/bin/glib-compile-schemas $out/share/glib-2.0/schemas
+    patchShebangs $out
+  '';
+
+  meta = with stdenv.lib; {
+    description = "NICE DCV is a high-performance remote display protocol";
+    homepage = "https://aws.amazon.com/hpc/dcv/";
+    license = licenses.unfree;
+    platforms = stdenv.lib.platforms.linux;
+    maintainers = [ maintainers.rmcgibbo ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/nice-dcv-client/default.nix:68]:
```suggestion
  meta = with stdenv.lib; {
    description = "NICE DCV is a high-performance remote display protocol";
    homepage = "https://aws.amazon.com/hpc/dcv/";
    license = licenses.unfree;
    platforms = platforms.linux;
    maintainers = with maintainers; [ rmcgibbo ];
```

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -7403,6 +7403,12 @@
     githubId = 13752145;
     name = "Richard Lupton";
   };
+  rmcgibbo = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:7406]: Please put this into a separate commit with the message ``maintainers: add rmcgibbo``.

File: pkgs/applications/networking/remote/nice-dcv-client/default.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
+{ stdenv, fetchurl, patchelf, bash,  glib
+  , libX11, gst_all_1, sqlite, epoxy, pango, cairo, gdk-pixbuf, e2fsprogs, libkrb5, libva, openssl
+  , pcsclite, gtk3, libselinux, libxml2
+  , python3Packages
+  , cpio
+}:
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/nice-dcv-client/default.nix:6]:
```suggestion
{ stdenv, fetchurl, patchelf, bash, glib
, libX11, gst_all_1, sqlite, epoxy, pango, cairo, gdk-pixbuf, e2fsprogs, libkrb5, libva, openssl
, pcsclite, gtk3, libselinux, libxml2, python3Packages, cpio
}:
```

File: pkgs/applications/networking/remote/nice-dcv-client/default.nix [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  buildCommand = ''
+    mkdir -p $out/bin/
+    libexecdir=usr/libexec/dcvviewer/
+
+    cd $out
+    rpm2cpio $src | ${cpio}/bin/cpio -idm
+
+    patchelf \
+      --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      $libexecdir/dcvviewer
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/nice-dcv-client/default.nix:28]:
```suggestion

    cd $out
    rpm2cpio $src | ${cpio}/bin/cpio -idm

    patchelf \
      --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
      usr/libexec/dcvviewer/dcvviewer
```

File: pkgs/applications/networking/remote/nice-dcv-client/default.nix [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+    mv $out/usr/bin/dcvviewer $out/bin/dcvviewer
+    sed -i "s#basedir=/usr#basedir=$out/usr#" $out/bin/dcvviewer
+    ln -s $out/usr/share $out/share
+    ${glib.dev}/bin/glib-compile-schemas $out/share/glib-2.0/schemas
+    patchShebangs $out
+  '';
+
+  meta = with stdenv.lib; {
+    description = "NICE DCV is a high-performance remote display protocol";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/nice-dcv-client/default.nix:64]:
```suggestion
    description = "High-performance remote display protocol";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by rmcgibbo** [comment]
Thanks for the review, @SuperSandro2000! I'll get this fixed up soon.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103939` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nice-dcv-client</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Review by markuskowa — CHANGES_REQUESTED** [review]

**Comment by rmcgibbo** [comment]
@markuskowa: good suggestion with autopatchelf. I've re-arranged how the phases work according, I think, to what you were suggesting and it's much more concise.

File: pkgs/applications/networking/remote/nice-dcv-client/default.nix [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  src =
+    fetchurl {
+      url = "https://d1uj6qtbmh3dt5.cloudfront.net/2020.2/Clients/nice-dcv-viewer-${version}.el8.x86_64.rpm";
+      sha256 = "sha256-SUpfHd/Btc07cfjc3zx5I5BiNatr/c4E2/mfJuU4R1E=";
+    };
+
+  nativeBuildInputs = [ autoPatchelfHook python3Packages.rpm ];
+  unpackPhase = ''
+    rpm2cpio $src | ${cpio}/bin/cpio -idm
+    echo "pwd = $(pwd)"
```
Comment (markuskowa) [inline pkgs/applications/networking/remote/nice-dcv-client/default.nix:37]: This looks like some left over debugging stuff?

Comment (rmcgibbo) [reply pkgs/applications/networking/remote/nice-dcv-client/default.nix]: yep, fixed and force-pushed.

**Review by markuskowa — APPROVED** [review]
